modules/collect/src/main/java/com/opengamma/collect/timeseries/LocalDateDoubleTimeSeries.java

```diff
@@ -1,75 +1,65 @@
- * 
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
  * Please see distribution for license.
  */
 package com.opengamma.collect.timeseries;
 
-import java.io.Serializable;
 import java.time.LocalDate;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Map;
-import java.util.Map.Entry;
 import java.util.NoSuchElementException;
 import java.util.OptionalDouble;
-import java.util.Set;
 import java.util.function.DoubleBinaryOperator;
 import java.util.function.DoubleUnaryOperator;
 import java.util.function.ObjDoubleConsumer;
 import java.util.stream.Collector;
-import java.util.stream.Collectors;
 import java.util.stream.DoubleStream;
-import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
-import org.joda.beans.Bean;
-import org.joda.beans.BeanBuilder;
-import org.joda.beans.BeanDefinition;
 import org.joda.beans.ImmutableBean;
-import org.joda.beans.ImmutableConstructor;
-import org.joda.beans.JodaBeanUtils;
-import org.joda.beans.MetaProperty;
-import org.joda.beans.Property;
-import org.joda.beans.PropertyDefinition;
-import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
-import org.joda.beans.impl.direct.DirectMetaBean;
-import org.joda.beans.impl.direct.DirectMetaProperty;
-import org.joda.beans.impl.direct.DirectMetaPropertyMap;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableSortedMap;
-import com.google.common.primitives.Doubles;
 import com.opengamma.collect.ArgChecker;
-import com.opengamma.collect.Guavate;
 import com.opengamma.collect.function.ObjDoublePredicate;
 
 /**
- * Standard immutable implementation of {@code DoubleTimeSeries}.
+ * Interface for all local date time-series types containing
+ * {@code double} values.
  * <p>
- * This implementation uses arrays internally.
+ * A time-series is similar to both a {@code SortedMap} of value keyed
+ * by date-time and a {@code List} of date-time to value pairs. As such,
+ * the date/times do not have to be evenly spread over time within the series.
+ * <p>
+ * The distribution of the data will influence which implementation
+ * is most appropriate.
+ * <p>
+ * Implementations must be immutable and thread-safe beans.
+ * <p>
+ * Note that {@link Double#NaN} is used internally as a sentinel
+ * value and is therefore not allowed as a value.
  */
-@BeanDefinition(builderScope = "private")
-public final class LocalDateDoubleTimeSeries
-    implements ImmutableBean, DoubleTimeSeries<LocalDate>, Serializable {
-
-  /** Empty instance. */
-  public static final LocalDateDoubleTimeSeries EMPTY_SERIES = createUnsafe(new LocalDate[0], new double[0]);
+public interface LocalDateDoubleTimeSeries extends ImmutableBean {
 
   /**
-   * The dates in the series.
-   * The dates are ordered from earliest to latest.
+   * Return an empty time-series. Generally a singleton instance
+   * is returned so this method can safely be called multiple times.
```

Comment:
Return -> Returns.
Also, "safely called multiple times" isn't exactly what you want to express here.